repository/src/main/java/org/pentaho/platform/repository2/userroledao/jackrabbit/security/DefaultPentahoPasswordEncoder.java

```diff
@@ -82,24 +82,20 @@ public boolean isPasswordValid( final String encPass, final String rawPass, fina
       return false;
     }
     try {
-      char[] decodedPassword = decodePassword( rawPass );
+      char[] decodedPassword = decodePassword( rawPass ).toCharArray();
       CryptedSimpleCredentials credentials = new CryptedSimpleCredentials( "dummyUser", encPass );
       return credentials.matches( new SimpleCredentials( "dummyUser", decodedPassword ) );
     } catch ( Exception e ) {
       throw new RuntimeException( e );
     }
   }
 
-  private static char[] decodePassword( String rawPass ) {
-    try {
-      if ( !StringUtils.isEmpty( rawPass ) && rawPass.startsWith( "ENC:" ) ) {
-        String password = new String( Base64Utils.fromBase64( rawPass.substring( 4 ) ), StandardCharsets.UTF_8 );
-        return URLDecoder.decode( password.replace( "+", "%2B" ), StandardCharsets.UTF_8.name() ).toCharArray();
-      } else {
-        return rawPass.toCharArray();
-      }
-    } catch ( UnsupportedEncodingException e ) {
-      return rawPass.toCharArray();
+  public static String decodePassword( String rawPass ) {
+    if ( !StringUtils.isEmpty( rawPass ) && rawPass.startsWith( "ENC:" ) ) {
+      String password = new String( Base64Utils.fromBase64( rawPass.substring( 4 ) ), StandardCharsets.UTF_8 );
+      return URLDecoder.decode( password.replace( "+", "%2B" ), StandardCharsets.UTF_8 );
+    } else {
+      return rawPass;
     }
   }
```

Comment:
If this method is to be used on both classes, it should exist on a third (independent) class.
The optimizations here, make total sense!